### PR TITLE
feat: catalog update topic command can now take multiple topics

### PR DESCRIPTION
**Pull request by lukebond**
This PR changes the `catalog topic update` subcommand to allow you to specify multiple topics. This will be used for the tool that syncs from the CRD list of topics, so that it only has to make one call.

### CODE REVIEW

**Comment by crepererum** [comment]
How would the CLI look like when we need additional per-topic parameters? I think we should rather call the CLI once per topic (we don't have that many topics anyways, so the overhead is OK).

File: influxdb_iox/src/commands/catalog/topic.rs [hunk 1]
```diff
@@ -43,10 +44,23 @@ enum Command {
...
+                .map(|n| async move { topics_repo.create_or_get(n).await })
+                .collect::<FuturesUnordered<_>>()
+                .collect::<Vec<_>>()
+                .await
+                .into_iter()
+                .collect::<Result<Vec<_>, _>>()?
+                .iter()
+                .for_each(|t| {
+                    println!("{}", t.id);
+                });
```
Comment (domodwyer) [inline influxdb_iox/src/commands/catalog/topic.rs:63]: If you import [`TryStream`](https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#) it handles `Result` nicely - in this example, [`try_for_each`](https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.try_for_each) might be helpful!

```suggestion
            update
                .topic_names
                .iter()
                .map(|n| async move { topics_repo.create_or_get(n).await })
                .collect::<FuturesUnordered<_>>()
                .try_for_each(|t| std::future::ready(Ok(println!("{}", t.id))))
                .await?;
```

**Review by domodwyer — COMMENTED** [review]
Thought this would be a good moment to introduce `TryStream` 🎉

File: influxdb_iox/src/commands/catalog/topic.rs (hunk 1 above)
Comment (lukebond) [reply influxdb_iox/src/commands/catalog/topic.rs:63]: much nicer :) thanks

**Comment by lukebond** [comment]
> How would the CLI look like when we need additional per-topic parameters? I think we should rather call the CLI once per topic (we don't have that many topics anyways, so the overhead is OK).

i hadn't thought of that, but i figured since the postgres schema has only name in the topic table it'd be okay? but yeah, expanding that table would break the tool that uses this CLI command i'm adding.

**Comment by crepererum** [comment]
> > How would the CLI look like when we need additional per-topic parameters? I think we should rather call the CLI once per topic (we don't have that many topics anyways, so the overhead is OK).
> 
> i hadn't thought of that, but i figured since the postgres schema has only name in the topic table it'd be okay? but yeah, expanding that table would break the tool that uses this CLI command i'm adding.

IMHO we should always be open to adding more columns to the table, esp. during this very early MVP phase. Looking at some of the tables I personally doubt that the design is final.

That said I'm not in favor of this interface, but I also don't veto it since YOU are both implementer and primary consumer. :slightly_smiling_face:

**Comment by lukebond** [comment]
I think you're right, I'll just loop at the other end! :+1:

**Pull request closed without merging**